### PR TITLE
CATL-1017: Support For Using Custom Contact Ref Fields To Autofill Contact In Webforms

### PR DESCRIPTION
**Pull request by vinuvarshith**
Overview
----------------------------------------
This ticket is targeted at improving the data loading capabilities of the ‘Existing Contact’ field of the Webform CiviCRM through the default contact setting
With this new setting - we will be able to preload the existing contact field with values derived from a particular custom field on the contact on the webform

Before
----------------------------------------
Currently we have the ability to load the existing contact fields of a contact on the webform with the following options:
Set default contact from:
Relationship to - Specify Contact - Specify Relationship

After
----------------------------------------
We have a new option in the default value options of the existing contact field on the webform.
sample

### CODE REVIEW

**Comment by shitijg** [comment]
@colemanw : https://www.drupal.org/project/webform_civicrm/issues/2996978 thanks

**Review by MiyaNoctem — COMMENTED** [review]
Look good to me.

**Comment by stesi561** [comment]
Works as advertised.

**Comment by shitijg** [comment]
@colemanw can we merge this?

File: includes/contact_component.inc [hunk 1]
```diff
@@ -238,6 +239,15 @@ function _webform_edit_civicrm_contact($component) {
       '#parents' => array('extra', 'default_relationship'),
     );
   }
+  $form['defaults']['default']['#options']['custom_ref'] = t('Custom reference field');
+  $form['defaults']['default_custom_ref'] = array(
+    '#type' => 'select',
+    '#title' => t('Specify Custom Field(s)'),
```
Comment (colemanw) [inline includes/contact_component.inc]: I believe the label "Field(s)" is incorrect. This is a select element not a multiselect, so you cannot pick more than one.

**Comment by colemanw** [comment]
@shitijg could you please fix the label and rebase this so it is mergable and then I'm happy to merge it. Thanks for the PR.

**Comment by vinuvarshith** [comment]
@colemanw @shitijg 
I have fixed the label and fixed conflicts

**Pull request merged**